packages/plugin/src/rules/naming-convention.ts

```diff
@@ -8,39 +8,78 @@ const formats = {
   UPPER_CASE: /^[A-Z_]*$/g,
 };
 
-function checkNameFormat(value, style, leadingUnderscore, trailingUnderscore) {
+type ValidNaming = 'camelCase' | 'PascalCase' | 'snake_case' | 'UPPER_CASE';
+const acceptedStyles: ValidNaming[] = ['camelCase', 'PascalCase', 'snake_case', 'UPPER_CASE'];
```

Comment:
You can define the `const` object as array of strings, and then create the `type ValidNaming` like that: `type ValidNaming = typeof acceptedStyles[number]`

Reply:
I have updated it in https://github.com/dotansimha/graphql-eslint/pull/201/commits/d337729c209478609c30f3a531c9bcefab6323fb. Hopefully that is what you meant :)

Reply:
Yeah, perfect :) Just to have a single place where the consts are defined.